starters/blog/gatsby-node.js

```diff
@@ -36,21 +36,25 @@ exports.createPages = async ({ graphql, actions, reporter }) => {
   const posts = result.data.allMarkdownRemark.nodes
 
   // Create blog posts pages
+  // But only if there's at least one markdown file found at "content/blog" (defined in gatsby-config.js)
   // `context` is available in the template as a prop and as a variable in GraphQL
-  posts.forEach((post, index) => {
-    const previous = index === posts.length - 1 ? null : posts[index + 1]
-    const next = index === 0 ? null : posts[index - 1]
-
-    createPage({
-      path: post.fields.slug,
-      component: blogPost,
-      context: {
-        slug: post.fields.slug,
-        previous,
-        next,
-      },
+
```

Comment:
This check shouldn't be needed, because if the array is empty, the forEach never happens anyway. However, what _might_ be needed is a null check on `allMarkdownRemark`. Does that exist if there's no markdown? Shame there's no optional chaining in gatsby-node!

Reply:
I prefer to make it more explicit as I myself can't always remember that e.g. `forEach` would behave this way